### DISCUSSION TITLE
TypeError: undefined is not a function with embla-carousel-react 8.0.0

### DISCUSSION THREAD

**Discussion by chemissov** [davidjerleke/embla-carousel#557] [Help] [answered]
I have a nextjs project and use embla 8.0.0-rc11.
The productive code is working well, but once I try to run a snapshot test with jest, I get this error:

undefined is not a function
TypeError: undefined is not a function
    at Array.map (<anonymous>)
    at optionsMediaQueries (/Users/.../node_modules/embla-carousel-react/embla-carousel-react.umd.js:1:13472)

none of my code files are in the stack trace, the error seems to happen in embla itself.

However, if I downgrade the lib to previous version 7.1.0 test passes through.

Any ideas, what is the problem and how can I deal with it (downgrade is not an option, since Im using some new features there...)

thx!

**Comment by davidjerleke** [comment, marked as answer]
Hi @chemissov,

Thank you for your question. You need to mock some stuff from v8 and up. [Read this](https://github.com/davidjerleke/embla-carousel/issues/516#issuecomment-1626013334) to understand it better.

Best,
David

**Reply by chemissov** [reply]
Great! Thank you for the fast answer. it worked 👍
